### PR TITLE
Completely Culls req_access_txt/req_one_access_txt

### LINKED ISSUE #72102: `text2access` still operates under the assumption accesses are numbers
## Reproduction:

https://github.com/tgstation/tgstation/blob/8bb6cb8f5c16b74af3b3150403284542c6fd3a9e/code/modules/jobs/access.dm#L56-L76

`text2access` converts a string access text into a list of accesses 

A long time ago, you would pass in a string like `"15;23;24"` you get `list(15, 23, 34)`
But now our accesses are strings, not numbers - If you set it up like `"robotics;science"`, you get `list()`

Some of our maps still use the `_txt` system to set accesses, namely on windoors, which means these are all incorrect. 

image

(Well, some of these are incorrect in another way. They use a list instead of a string. Wrong var)

### PR DESCRIPTION
**Pull request by san7890**
## About The Pull Request

Hey there,

Now that every instance of `req_access` and `req_one_access` is a list of strings, there is absolutely no reason for req_access_txt/req_access_one_txt to exist. In fact, any instance where they were still used in the codebase was very convoluted and was very broken! Don't worry, I fixed it all out, and life is good. 

I also dmdoc the surviving access variables, because those were missing. I went on the side of caution and made a more verbose documentation with an example just to have people really grasp this (it took me a while to actually get it)
## Why It's Good For The Game

Fixes #72102.

I believe that we changed _everything_ over to the req_access/req_one_access system earlier this year in VV, but the problem is that _new mappers don't understand the difference between the two systems_. In fact, the "txt" system is completely redundant since all it does is transition stuff to the "base" system. So, let's just completely cull the one that's all but deprecated and ensure this confusion no longer arises. The whole purpose of "txt" seemed to be to convert the access, but it's all pointless now that we can just read the list directly.

I'm also 99% certain that the "access check" on vending machines broke (and didn't seem to have correct logic in the first place? I legitimately couldn't find a case where it could fail in testing, so I changed that up), and that's fixed up now. Let me know if I was clueless there. I know it's short-circuiting now as opposed to "all must be true", but it just didn't work.
## Changelog
:cl:
fix: Those with kitchen access should now once again be the only one who are allowed to vending-machine-restock individual snacking items via their trusty tray.
/:cl:

### CODE REVIEW

File: code/modules/vending/_vending.dm [hunk 1]
```diff
@@ -169,10 +169,6 @@
 	///A variable to change on a per instance basis on the map that allows the instance to force cost and ID requirements
 	var/onstation_override = FALSE //change this on the object on the map to override the onstation check. DO NOT APPLY THIS GLOBALLY.
 
-	///ID's that can load this vending machine wtih refills
-	var/list/canload_access_list
```
Comment (san7890) [inline code/modules/vending/_vending.dm:173]: redundant

File: code/modules/vending/_vending.dm [hunk 2]
```diff
@@ -806,30 +802,20 @@
...
 /obj/machinery/vending/proc/compartmentLoadAccessCheck(mob/user)
-	if(!canload_access_list)
+	if(!req_access)
 		return TRUE
-	else
-		var/do_you_have_access = FALSE
-		var/req_access_txt_holder = req_access_txt
-		for(var/i in canload_access_list)
-			req_access_txt = i
-			if(!allowed(user) && !(obj_flags & EMAGGED) && scan_id)
```
Comment (san7890) [inline code/modules/vending/_vending.dm:822]: i might have been doing it wrong but i couldn't trigger this to work previously. it probably worked three years ago, but the modern framework seems to have completely passed this by.

File: code/modules/jobs/access.dm [hunk 3]
```diff
@@ -53,35 +53,11 @@
...
-	if(!access_text)
-		return
-	var/list/split = splittext(access_text,";")
-	for(var/x in split)
-		var/n = text2num(x)
-		if(n)
-			. += n
-
-//Call this before using req_access or req_one_access directly
-/obj/proc/gen_access()
```
Comment (san7890) [inline code/modules/jobs/access.dm:67]: all this proc did was convert things from the `txt` system to the "base" system

File: code/modules/jobs/access.dm [hunk 4]
```diff
@@ -53,35 +53,11 @@
 /obj/item/proc/InsertID()
 	return FALSE
 
-/obj/proc/text2access(access_text)
```
Comment (san7890) [inline code/modules/jobs/access.dm:56]: this isn't needed now that everything access-wise is a nice list()

File: code/game/objects/objs.dm [hunk 5]
```diff
@@ -29,9 +29,8 @@
 
 	// Access levels, used in modules\jobs\access.dm
 	var/list/req_access
-	var/req_access_txt = "0"
 	var/list/req_one_access
-	var/req_one_access_txt = "0"
```
Comment (san7890) [inline code/game/objects/objs.dm:34]: THE BEAST IS DEAD

**Comment by Capsandi** [comment]
End of an era 😔

**Comment by JohnFulpWillard** [comment]
Thank you for this, it was really confusing when I first started mapping, you're making the lives of the future generations that much better.

**Pull request merged**